source/guides/rapids-docker-with-cuda.md

```diff
@@ -0,0 +1,371 @@
+    cuda-libraries-12-9=${NV_CUDA_LIB_VERSION} \
+    ${NV_LIBNPP_PACKAGE} \
+    cuda-nvtx-12-9=${NV_NVTX_VERSION} \
+    libcusparse-12-9=${NV_LIBCUSPARSE_VERSION} \
+    ${NV_LIBCUBLAS_PACKAGE} \
+    ${NV_LIBNCCL_PACKAGE} && \
+    rm -rf /var/lib/apt/lists/*
+
+# Keep apt from auto upgrading the cublas and nccl packages
+RUN apt-mark hold ${NV_LIBCUBLAS_PACKAGE_NAME} ${NV_LIBNCCL_PACKAGE_NAME}
+
+# CUDA Environment Configuration
+ENV PATH=/usr/local/cuda/bin:${PATH}
+ENV LD_LIBRARY_PATH=/usr/local/cuda/lib64
+
+# NVIDIA Container Runtime Configuration
+ENV NVIDIA_VISIBLE_DEVICES=all
+ENV NVIDIA_DRIVER_CAPABILITIES=compute,utility
+
+# Required for nvidia-docker v1
+RUN echo "/usr/local/cuda/lib64" >> /etc/ld.so.conf.d/nvidia.conf
+
+# Install system dependencies
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    python${PYTHON_VER} \
+    python${PYTHON_VER}-venv \
+    python3-pip \
+    wget \
+    curl \
+    git \
+    ca-certificates \
+    && rm -rf /var/lib/apt/lists/*
+
+# Create symbolic links for python and pip
+RUN ln -sf /usr/bin/python${PYTHON_VER} /usr/bin/python && \
+    ln -sf /usr/bin/python${PYTHON_VER} /usr/bin/python3
+
+# Create rapids user
+RUN groupadd -g 1001 rapids && \
+    useradd -rm -d /home/rapids -s /bin/bash -g rapids -u 1001 rapids
+
+USER rapids
+WORKDIR /home/rapids
+
+# Create and activate virtual environment
+RUN python -m venv /home/rapids/venv
+ENV PATH="/home/rapids/venv/bin:$PATH"
+
+# Upgrade pip
+RUN pip install --no-cache-dir --upgrade pip setuptools wheel
+
+# Copy the requirements file
+COPY requirements.txt /home/rapids/requirements.txt
+
+# Install all packages
+RUN pip install --no-cache-dir -r requirements.txt
+
+CMD ["bash"]
+```
+
+## Verifying Your Installation
+
+After starting your container, you can quickly test that RAPIDS is installed and running correctly. The container launches directly into a `bash` shell where you can install the [RAPIDS CLI](https://github.com/rapidsai/rapids-cli) command line utility to verify your installation.
+
+1. **Run the Container Interactively**
+
+   This command starts your container and drops you directly into a bash shell.
+
+   ```bash
+   # Build the conda-based container (requires env.yaml in build context)
+   docker build -f conda-rapids.Dockerfile -t rapids-conda-cuda .
+
+   # Build the pip-based container (requires requirements.txt in build context)
+   docker build -f pip-rapids.Dockerfile -t rapids-pip-cuda .
+
+   # Run conda container with GPU access
+   docker run --gpus all -it rapids-conda-cuda
+
+   # Run pip container with GPU access
+   docker run --gpus all -it rapids-pip-cuda
+   ```
+
+2. **Install RAPIDS CLI**
+
+   Inside the containers, install the RAPIDS CLI:
+
+   ```bash
+   pip install rapids-cli
+   ```
+
+3. **Test the installation using the Doctor subcommand**
+
+   Once RAPIDS CLI is installed, you can use the `rapids doctor` subcommand to perform health checks.
+
+   ```bash
```

Comment:
we can use console and show the desired output, it would be like this. 

```console
$ rapids doctor
🧑‍⚕️ Performing REQUIRED health check for RAPIDS
Running checks
All checks passed!
```